### PR TITLE
add tests for Optimizely coverage

### PR DESCRIPTION
**Pull request by jaeopt**

### CODE REVIEW

File: OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_Invalid.swift [hunk 1]
```diff
@@ -116,3 +127,19 @@ class OptimizelyClientTests_Invalid: XCTestCase {
     }
 
 }
+
+// MARK: - Invalid Datafile
+
+extension OptimizelyClientTests_Invalid {
+    
+//    func testActivate_WhenManagerNonInitialized() {
+//        do {
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_Invalid.swift]: what's this?

File: OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_ObjcOthers.m [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+- (void)flushEvents {
+    return;
+}
+@end
+
+@implementation OptimizelyClientTests_ObjcOthers
+
+// MARK: - Others
+
+- (void)testCustomNotificationCenter_Activate {
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_ObjcOthers.m]: this seems incorrectly named CustomNotificationCenter_Activate?

File: OptimizelySDK/Customization/DefaultEventDispatcher.swift [hunk 3]
```diff
@@ -75,9 +75,9 @@ open class DefaultEventDispatcher: BackgroundingCallbacks, OPTEventDispatcher {
     open func dispatchEvent(event: EventForDispatch, completionHandler: DispatchCompletionHandler?) {
         dataStore.save(item: event)
         
-        // TODO: use or clean up completionHandler
-
         setTimer()
+        
+        completionHandler?(.success(event.body))
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/Customization/DefaultEventDispatcher.swift]: this doesn't seem like the correct approach.  maybe something  better is to store a list of all callbacks without a retain and then call them on anytime an event is actually sent.

Comment (jaeopt) [reply OptimizelySDK/Customization/DefaultEventDispatcher.swift]: The current way of calling back on queueing makes sense to me:
- We do not have any case for queueing failure for now, but this call can be used for notification of "event queued successfully and it'll be delivered ok"
- Event delivery can be delayed arbitrary time. Keeping all callback handlers may be expensive and error-prone.

File: OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_Invalid.swift (hunk 1 above)
Comment (jaeopt) [reply OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_Invalid.swift]: removed

File: OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_ObjcOthers.m (hunk 2 above)
Comment (jaeopt) [reply OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_ObjcOthers.m]: The name "testCustomNotificationCenter_Activate" looks ok to me. Not clear what you mean here.

Comment (thomaszurkan-optimizely) [reply OptimizelySDK/OptimizelyTests/OptimizelyTests-APIs/OptimizelyClientTests_ObjcOthers.m]: there is no custom notification center.  it should be testNotificationCenter_Activate

**Pull request merged**